express-post-json/index.js

```diff
@@ -0,0 +1,29 @@
+const express = require('express');
+const app = express();
+
+let nextId = 1;
+const grades = {};
+
+app.use(express.json());
+
+app.get('/api/grades', (req, res) => {
+  const gradesArray = [];
+  for (const id in grades) {
+    gradesArray.push(grades[id]);
+  }
+  res.json(gradesArray);
+});
+
+app.post('/api/grades', (req, res) => {
+  const newGrade = req.body;
+  newGrade.id = nextId;
+  grades[nextId] = newGrade;
+  nextId++;
+  res.send(newGrade);
+  res.status(201);
```

Comment:
If you set the status code after sending the response, it will not have any effect. Instead, set the status code before `res.send()`.